catalog/video/video-mpegts-stream/src/test/java/org/codice/alliance/video/stream/mpegts/plugins/FrameCenterUpdateFieldTest.java

```diff
@@ -0,0 +1,131 @@
+        when(childMetacard2.getAttribute(AttributeNameConstants.FRAME_CENTER)).thenReturn(new AttributeImpl(
+                AttributeNameConstants.FRAME_CENTER,
+                wktChild2));
+        Metacard childMetacard3 = mock(Metacard.class);
+        when(childMetacard3.getAttribute(AttributeNameConstants.FRAME_CENTER)).thenReturn(new AttributeImpl(
+                AttributeNameConstants.FRAME_CENTER,
+                wktChild3));
+
+        FrameCenterUpdateField frameCenterUpdateField =
+                new FrameCenterUpdateField(GeometryOperator.IDENTITY, new GeometryFactory());
+
+        Context context = mock(Context.class);
+        GeometryOperator.Context geometryOperatorContext = new GeometryOperator.Context();
+        geometryOperatorContext.setSubsampleCount(1000);
+        when(context.getGeometryOperatorContext()).thenReturn(geometryOperatorContext);
+
+        frameCenterUpdateField.updateField(parentMetacard,
+                Collections.singletonList(childMetacard1),
+                context);
+        frameCenterUpdateField.updateField(parentMetacard,
+                Arrays.asList(childMetacard2, childMetacard3),
+                context);
+        frameCenterUpdateField.end(parentMetacard, context);
+
+        ArgumentCaptor<Attribute> captor = ArgumentCaptor.forClass(Attribute.class);
+
+        verify(parentMetacard).setAttribute(captor.capture());
+
+        assertThat(captor.getValue()
+                .getValue(), is(wktExpected));
+
+    }
+
+    @Test
+    public void testThatGeoOperatorIsCalled() throws ParseException {
+
+        String wktChild1 = "LINESTRING (30 10, 10 30, 40 40)";
+
+        GeometryOperator geometryOperator = mock(GeometryOperator.class);
+        Geometry geometry = new WKTReader().read("LINESTRING (0 0, 1 1)");
+
+        Context context = mock(Context.class);
+
+        GeometryOperator.Context geometryOperatorContext = new GeometryOperator.Context();
+        geometryOperatorContext.setSubsampleCount(1000);
+
+        when(context.getGeometryOperatorContext()).thenReturn(geometryOperatorContext);
+        when(geometryOperator.apply(any(), any())).thenReturn(geometry);
+
+        Metacard parentMetacard = mock(Metacard.class);
+
+        Metacard childMetacard1 = mock(Metacard.class);
+        when(childMetacard1.getAttribute(AttributeNameConstants.FRAME_CENTER)).thenReturn(new AttributeImpl(
+                AttributeNameConstants.FRAME_CENTER,
+                wktChild1));
+
+        FrameCenterUpdateField frameCenterUpdateField = new FrameCenterUpdateField(geometryOperator,
+                new GeometryFactory());
+
+        frameCenterUpdateField.updateField(parentMetacard,
+                Collections.singletonList(childMetacard1),
+                context);
+
+        verify(geometryOperator, times(0)).apply(any(), any());
+
+        frameCenterUpdateField.end(parentMetacard, context);
+
+        verify(geometryOperator, times(1)).apply(any(), any());
+
+    }
+
```

Comment:
I added an extra unit test for the case with various types of invalid frame center attribute values.